sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubCoderProviderRegistrar.java

```diff
@@ -31,6 +31,7 @@ public class PubsubCoderProviderRegistrar implements CoderProviderRegistrar {
   @Override
   public List<CoderProvider> getCoderProviders() {
     return ImmutableList.of(
+        CoderProviders.forCoder(TypeDescriptor.of(PubsubMessage.class), PubsubMessageCoder.of()),
```

Comment:
I'm not 100% confident in the order that is required here, or if order matters.  I couldn't find documentation around how coders are resolved when multiple coders are registered for the same class.  I tried tracing the code but wasn't able to come up with a definitive answer